duo_universal/client.py

```diff
@@ -198,21 +207,23 @@ def health_check(self):
 
         return res
 
-    def create_auth_url(self, username, state):
+    def create_auth_url(self, username, state, nonce=None):
         """Generate uri to Duo's prompt
 
         Arguments:
 
         username        -- username trying to authenticate with Duo
         state           -- Randomly generated character string of at least 22
                            chars returned to the integration by Duo after 2FA
+        nonce           -- Randomly generated character string of at least 16
```

Comment:
You say 16 characters here, but it looks like you're reusing the state parameter's limits (back at line 103), so is 16 correct, or the 22 that line 216 has?

Reply:
I believe this SDK is backed by the Duo OIDC API, and in the docs (https://duo.com/docs/oauthapi#authorization-request) both the nonce:

" Optional 	If provided, this must be a random value that is cryptographically secure such that it is unguessable by a third party. We require a length of at least 16 characters and at most 1024 characters. "

and state: 

"A random, cryptographically secure value that the client must maintain throughout the authentication. We require a length of at least 16 characters and at most 1024 characters. "

appear to specify a length of 16 to 1024 characters. Perhaps the thing to do is to also have the documentation for state also reflect these limitations?

Reply:
I agree.... this is the first SDK we wrote so maybe things hadn't been finalized yet.  I'll open an issue to check and correct the various limits.

Reply:
Issue 23 opened